From c38ec6078b3bf2a5d151d1b61b2efa21615a7a06 Mon Sep 17 00:00:00 2001
From: Cody Peter Mello <cody.mello@joyent.com>
Date: Fri, 29 Jun 2018 17:42:31 +0000
Subject: [PATCH] TRITON-563 net-agent dumps core due to attempt to change
 child logger's name Reviewed by: Alex Wilson <alex.wilson@joyent.com>
 Approved by: Alex Wilson <alex.wilson@joyent.com>

---
 lib/aggr-fsm.js   | 2 +-
 lib/server-fsm.js | 1 +
 2 files changed, 2 insertions(+), 1 deletion(-)

diff --git a/lib/aggr-fsm.js b/lib/aggr-fsm.js
index 04490e4..a874297 100644
--- a/lib/aggr-fsm.js
+++ b/lib/aggr-fsm.js
@@ -86,7 +86,7 @@ function AggrFSM(opts) {
     this.app = opts.app;
     this.log = opts.app.log.child({
         component: 'aggr',
-        name: this.name
+        aggr: this.name
     }, true);
 
     this.pending = {
diff --git a/lib/server-fsm.js b/lib/server-fsm.js
index 9fb09aa..7019b98 100644
--- a/lib/server-fsm.js
+++ b/lib/server-fsm.js
@@ -43,6 +43,7 @@ function ServerFSM(opts) {
     }, true);
 
     this.nics = {};
+    this.aggrs = {};
     this.nictags = {};
 
     mod_mooremachine.FSM.call(this, 'init');
-- 
2.21.0

